ai/docs/GPT Online Model Fine-tuning Documentation.md

```diff
@@ -0,0 +1,332 @@
+# GPT Online Model Fine-tuning Documentation
+## Introduction
+GPT is a large-language-model that can give response to your questions in natural language, by calculating the representation of your questions with its model parameters. However, for customized scenarios, the original models OpenAI releases may not work very well. If you come across this problem, model fine-tuning may be a good solution. The working principle of fine-tuning is using your data which you want the model to learn, and adjusting the model parameters by back propagation algorithm with your data. It makes AI model following instructions better, as well as formatting output responses.
+
+We have re-encapsulated the OpenAI model fine-tuning interface. This document describes how to use your own data to finetune an OpenAI GPT online model, just by following 3 steps: data preparing, fine-tuning Job creating, and fine-tuned model calling. 
+
+Before the fine-tuning process, you should have **Python3 Environment** installed, and the **openai, pathlib** Python package installed. Up to now, it can run with the latest version of Python3 and the packages, and you could try other versions. For users already have Python3 and pip, you can use the following command to install packages in requirements.txt (same folder with the scripts).
+```plain
+pip install -r requirements.txt
+```
+Also, you should have an **OpenAI api-key** to run the fine-tuning jobs.
+All the python scripts' usage can be shown with the command:
+```plain
+python3 <python_script> -h
+```
+
+For OpenAI official tutorial, please [refer to this](https://openai.com/blog/gpt-3-5-turbo-fine-tuning-and-api-updates), or [more specific one](https://platform.openai.com/docs/guides/fine-tuning).
+
+## Steps
+
+1. Prepare formatted data
```

Comment:
There is no line break In the real display.

Reply:
updated